### PR TITLE
Admin vs Non-Admin Usage & Settings UX Update

### PR DESCRIPTION
**Pull request by ShadeWyrm**
This PR is a large chunk of the requirements from #3313

It is split off from #3647 to avoid issues with Salesforce Integration.

---

For Testing...

The UI/UX changes are on the Form Settings Page, and Publish page.

Publish Page:
* The User can no longer publish if the Admin vs Non-Admin use is set to Unset. (New Form)
* The User will get a happy checkmark that they can publish if it's set.

Settings Page:
* The user can pick between the two options for Administrative vs Non-Administrative Use of a form. This value is saved to the template store and by proxy the db.

The pages should work the same and have valid relevant text in both languages.

### CODE REVIEW

**Comment by timarney** [comment]
@ShadeWyrm can you please add some test instructions to the PR description so folks without context can review without too much digging on what the PR does ?

**Comment by timarney** [comment]
@ShadeWyrm this is looking good 

Thinking we can remove the getter / setter for the `useTemplateStore`  i.e.  https://github.com/cds-snc/platform-forms-client/commit/3af5a8e96726b3eaa0c18989a57300be6aeedcdb does that work?

File: app/(gcforms)/[locale]/(form administration)/form-builder/[id]/settings/components/ResponseDelivery.tsx [hunk 1]
```diff
@@ -33,6 +33,12 @@ enum DeliveryOption {
   email = "email",
 }
 
+export enum PurposeOption {
```
Comment (timarney) [inline app/(gcforms)/[locale]/(form administration)/form-builder/[id]/settings/components/ResponseDelivery.tsx:36]: Thinking we should start commenting these to make it easier to folks looking at these types

i.e. something like 


```
/*
  * PurposeOption is used to determine the purpose of the form
  * admin: The form is used to collect personal information
  * nonAdmin: The form is used to collect non-personal information
  */
```



I know these don't exist for Vault etc... but might make it easier to folks onboarding or who haven't touched this part of the code.

**Comment by ShadeWyrm** [comment]
> @ShadeWyrm this is looking good
> 
> Thinking we can remove the getter / setter for the `useTemplateStore` i.e. [3af5a8e](https://github.com/cds-snc/platform-forms-client/commit/3af5a8e96726b3eaa0c18989a57300be6aeedcdb) does that work?
> 
> Get
> 
> ```js
>  formPurpose: s.formPurpose,
> ```
> 
> Update
> 
> ```js
>  updateField("formPurpose", purposeOption);
> ```

Yah that should work just fine, it was a hold over from my testing, and figuring out the template store. I had assumed it was necessary, but the Linter brought that into question on my end. :) So just the property should be enough.

File: app/(gcforms)/[locale]/(form administration)/form-builder/[id]/settings/components/ResponseDelivery.tsx (hunk 1 above)
Comment (ShadeWyrm) [reply app/(gcforms)/[locale]/(form administration)/form-builder/[id]/settings/components/ResponseDelivery.tsx:36]: Agreed.

File: lib/tests/templates.test.ts [hunk 2]
```diff
@@ -65,12 +65,14 @@ const buildPrismaResponse = (
   jsonConfig: object,
   isPublished = false,
   deliveryOption?: DeliveryOption,
+  formPurpose?: string,
```
[the remark this answers is not in the document]
Comment (ShadeWyrm) [reply lib/tests/templates.test.ts:68]: Probably makes sense to do so, you're right.

**Comment by timarney** [comment]
https://github.com/cds-snc/platform-forms-client/pull/3714#issuecomment-2142090035

Do you want to cherry pick that commit in or make the updates on you end?

**Review by timarney — APPROVED** [review]
Nice work :)

**Pull request merged**